frontend/src/components/ui/input.tsx

```diff
@@ -155,4 +155,31 @@ export const SearchInput = React.forwardRef<
 });
 SearchInput.displayName = "SearchInput";
 
+export const OnBlurredInput = React.forwardRef<
+  HTMLInputElement,
+  InputProps & {
+    value: string;
+    onValueChange: (value: string) => void;
+  }
+>(({ className, onValueChange, ...props }, ref) => {
+  const [internalValue, setInternalValue] = React.useState(props.value);
```

Comment:
this might get out of sync when there is another blurred text input. 
you might need to make a similar `useControlledState` as we do with the debounce above. you might be able to leverage `useControllableState` from radix (its used somewhere else as well)